### PR TITLE
Foundations: Update descriptive links (#27681)

### PR DESCRIPTION
**Pull request by yuliana-r**
## Because
Update descriptive links to improve accessibility.

## This PR
- updates the links in [Object Basics](https://www.theodinproject.com/lessons/foundations-object-basics)
- updates the links in [Project: Calculator](https://www.theodinproject.com/lessons/foundations-calculator)
- updates the links in [Choose Your Path Forward](https://www.theodinproject.com/lessons/foundations-choose-your-path-forward)

## Issue
Related to #27681

## Additional Information
N/a

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for getting these links updated, @yuliana-r.

Would you be able to fix the remaining lint issues? Some of them have come from what presumably might have been an automatic formatter on your local machine? e.g. changing `*` emphasis to `_` emphasis - these will need changing back to `*`.
Some of the lint errors were already there but they should be straightforward to fix, if you wouldn't mind knocking them out as well?

If you are using a formatter and especially if you have it auto-format on save, you'll likely find it much easier to either make a workspace profile on VSCode that doesn't format when in the curriculum repo as not all of the rules the formatter would use will match our style guide. Or you can bind formatting to a keyboard shortcut instead of formatting on save, and then just not hit that keybind when working on the curriculum repo.

Regarding the TOP004 heading error in the Conclusion lesson on line 17, that can be remedied by adding the following comments to the top of that file, as in this scenario, we want to manually disable that rule for this lesson:
```html
<!-- TOP004 disabled due to unique and intended lesson structure  -->
<!-- markdownlint-disable TOP004 -->
```

Let me know if you have any issues.

File: foundations/javascript_basics/project_calculator.md [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
-    - Display a snarky error message if the user tries to divide by 0... and don't let it crash your calculator!
+   - **Your calculator should not evaluate more than a single pair of numbers at a time.** Example: you press a number button (`12`), followed by an operator button (`+`), a second number button (`7`), and finally a second operator button (`-`). Your calculator should then do the following: first, evaluate the first pair of numbers (`12 + 7`), second, display the result of that calculation (`19`), and finally, use that result (`19`) as the first number in your new calculation, along with the next operator (`-`). An example of the behavior we’re looking for can be seen in this [student's calculator live preview](https://mrbuddh4.github.io/calculator/).
+   - You should round answers with long decimals so that they don't overflow the screen.
+   - Pressing `=` before entering all of the numbers or an operator could cause problems!
+   - Pressing "clear" should wipe out any existing data.. make sure the user is really starting fresh after pressing "clear"
```
Comment (MaoShizhong) [inline foundations/javascript_basics/project_calculator.md:42]:
```suggestion
   - Pressing "clear" should wipe out any existing data. Make sure the user is really starting fresh after pressing "clear".
```
Just noticed this pre-existing punctuation issue, if you wouldn't mind throwing this fix in as well?

**Comment by yuliana-r** [comment]
My bad about the formatting - won't happen in the future. Thank you :)

File: foundations/tying_it_all_together/conclusion.md [hunk 2]
```diff
@@ -54,6 +57,6 @@ The journey has just begun!
 
 ### Additional resources
 
-This section contains helpful links to related content. It isn’t required, so consider it supplemental.
+This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
-- If you are still confused, Check out this article to get more facts, pros and cons about both languages [Ruby on Rails vs Javascript.](https://cloudinfrastructureservices.co.uk/ruby-on-rails-vs-javascript-whats-the-difference/).
+- If you are still confused, Check out this article to get more facts, pros and cons about both languages [Ruby on Rails vs JavaScript.](https://cloudinfrastructureservices.co.uk/ruby-on-rails-vs-javascript-whats-the-difference/).
```
Comment (MaoShizhong) [inline foundations/tying_it_all_together/conclusion.md:62]:
```suggestion
- If you are still confused, Check out this article to get more facts, pros and cons about both languages [Ruby on Rails vs JavaScript](https://cloudinfrastructureservices.co.uk/ruby-on-rails-vs-javascript-whats-the-difference/).
```
Not something you introduced but just noticed this in the lesson preview tool.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for doing that! 
I'm also an idiot and completely missed one link and didn't add it to the spreadsheet.

`object_basics.md` line 188:
```text
ask for help on our [Discord](https://discord.gg/fbFCkYabZB)!
```
Could this snippet be changed to the following?
```text
ask for help in the [TOP Discord server](https://discord.com/invite/theodinproject)!
```

**Review by MaoShizhong — APPROVED** [review]
Thank you again! :rocket:

**Pull request merged**